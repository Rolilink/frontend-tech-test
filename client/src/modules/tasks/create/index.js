import connectToState from './providers/CreateTask';
import CreateTaskComponent from './components/CreateTask';

export default connectToState(CreateTaskComponent);
