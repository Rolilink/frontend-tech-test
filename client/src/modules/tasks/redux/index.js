import * as actionTypes from './actionTypes';
import * as factories from './factories';
import * as selectors from './selectors';
import * as utils from './utils';

import { fetchTasks, createTask } from './actions';

import reducerModule from './reducers';

const reducer = {
  name: 'tasks', reducer: reducerModule,
};

export {
  fetchTasks,
  createTask,
  actionTypes,
  factories,
  selectors,
  reducer,
  utils,
};
