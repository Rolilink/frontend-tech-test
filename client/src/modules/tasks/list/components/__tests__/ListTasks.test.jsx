import React from 'react';
import { shallow, mount } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import ShowTasks from '../ListTasks';
import { taskFactory } from '../../../../../__tests__/testUtils/tasks';

expect.addSnapshotSerializer(serializer);

const tasks = [
  taskFactory({
    id: 1,
    title: 'Task Title',
    description: 'Task Description',
  }),
];

describe('Tasks.List.ListTasks', () => {
  it('should call fetchTasks prop when mounting', () => {
    const fetchTasks = jest.fn();

    shallow(<ShowTasks fetchTasks={fetchTasks} />);

    expect(fetchTasks.mock.calls.length).toBe(1);
  });

  it('should render tasks when a tasks object array is passed through the tasks prop', () => {
    const fetchTasks = jest.fn();

    const showTasks = mount(<ShowTasks fetchTasks={fetchTasks} tasks={tasks} />);

    expect(showTasks).toMatchSnapshot();
  });

  it('should render an error message when an error message is passed through error prop', () => {
    const fetchTasks = jest.fn();

    const showTasks = mount(<ShowTasks fetchTasks={fetchTasks} error="error" />);

    expect(showTasks).toMatchSnapshot();
  });

  it('should render a loading message when the isFetching prop is present', () => {
    const fetchTasks = jest.fn();

    const showTasks = mount(<ShowTasks fetchTasks={fetchTasks} isFetching />);

    expect(showTasks).toMatchSnapshot();
  });

  it('should render an "there is no tasks" message when tasks prop is empty and isFetching prop is not present', () => {
    const fetchTasks = jest.fn();

    const showTasks = mount(<ShowTasks fetchTasks={fetchTasks} />);

    expect(showTasks).toMatchSnapshot();
  });

  it('#getErrorMessage should return a error message', () => {
    const fetchTasks = jest.fn();

    const showTasks = shallow(<ShowTasks fetchTasks={fetchTasks} />);

    expect(showTasks.instance().getErrorMessage('message')).toMatchSnapshot();
  });

  it('#getLoadingMessage should return a loading message', () => {
    const fetchTasks = jest.fn();

    const showTasks = shallow(<ShowTasks fetchTasks={fetchTasks} />);

    expect(showTasks.instance().getLoadingMessage()).toMatchSnapshot();
  });

  it('#getEmptyMessage should return a message', () => {
    const fetchTasks = jest.fn();

    const showTasks = shallow(<ShowTasks fetchTasks={fetchTasks} />);

    expect(showTasks.instance().getEmptyMessage()).toMatchSnapshot();
  });
});
