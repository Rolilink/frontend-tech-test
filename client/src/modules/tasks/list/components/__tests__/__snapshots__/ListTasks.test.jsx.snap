// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks.List.ListTasks #getEmptyMessage should return a message 1`] = `
<div>
  No Tasks Found
</div>
`;

exports[`Tasks.List.ListTasks #getErrorMessage should return a error message 1`] = `
<div>
  message
</div>
`;

exports[`Tasks.List.ListTasks #getLoadingMessage should return a loading message 1`] = `
<div>
  Loading Tasks
</div>
`;

exports[`Tasks.List.ListTasks should render a loading message when the isFetching prop is present 1`] = `
<ListTasks
  className=""
  error={null}
  fetchTasks={[Function]}
  isFetching={true}
  tasks={Array []}
>
  <LoadingMessage
    className=""
    isLoading={true}
    renderMessage={[Function]}
  >
    <div
      className="LoadingMessage "
    >
      <div
        className="LoadingMessage-Spinner"
      >
        <Spinner>
          <i
            className="fa fa-spinner fa-spin"
          />
        </Spinner>
      </div>
      <div
        className="LoadingMessage-Message"
      >
        <div>
          Loading Tasks
        </div>
      </div>
    </div>
  </LoadingMessage>
</ListTasks>
`;

exports[`Tasks.List.ListTasks should render an "there is no tasks" message when tasks prop is empty and isFetching prop is not present 1`] = `
<ListTasks
  className=""
  error={null}
  fetchTasks={[Function]}
  isFetching={false}
  tasks={Array []}
>
  <LoadingMessage
    className=""
    isLoading={false}
    renderMessage={[Function]}
  >
    <ErrorMessage
      className=""
      error={null}
      renderMessage={[Function]}
    >
      <EmptyMessage
        className=""
        isEmpty={true}
        renderMessage={[Function]}
      >
        <div
          className="EmptyMessage "
        >
          <div
            className="EmptyMessage-Sign"
          >
            Here goes an empty sign
          </div>
          <div
            className="EmptyMessage-Message"
          >
            <div>
              No Tasks Found
            </div>
          </div>
        </div>
      </EmptyMessage>
    </ErrorMessage>
  </LoadingMessage>
</ListTasks>
`;

exports[`Tasks.List.ListTasks should render an error message when an error message is passed through error prop 1`] = `
<ListTasks
  className=""
  error="error"
  fetchTasks={[Function]}
  isFetching={false}
  tasks={Array []}
>
  <LoadingMessage
    className=""
    isLoading={false}
    renderMessage={[Function]}
  >
    <ErrorMessage
      className=""
      error="error"
      renderMessage={[Function]}
    >
      <div
        className="ErrorMessage "
      >
        <div
          className="ErrorMessage-Sign"
        >
          Here goes an error sign
        </div>
        <div
          className="ErrorMessage-Message"
        >
          <div>
            error
          </div>
        </div>
      </div>
    </ErrorMessage>
  </LoadingMessage>
</ListTasks>
`;

exports[`Tasks.List.ListTasks should render tasks when a tasks object array is passed through the tasks prop 1`] = `
<ListTasks
  className=""
  error={null}
  fetchTasks={[Function]}
  isFetching={false}
  tasks={
    Array [
      Object {
        "description": "Task Description",
        "id": 1,
        "title": "Task Title",
      },
    ]
  }
>
  <LoadingMessage
    className=""
    isLoading={false}
    renderMessage={[Function]}
  >
    <ErrorMessage
      className=""
      error={null}
      renderMessage={[Function]}
    >
      <EmptyMessage
        className=""
        isEmpty={false}
        renderMessage={[Function]}
      >
        <TasksList
          className="ShowTasks-TasksList "
          tasks={
            Array [
              Object {
                "description": "Task Description",
                "id": 1,
                "title": "Task Title",
              },
            ]
          }
        >
          <List
            className="TasksList ShowTasks-TasksList "
            items={
              Array [
                Object {
                  "description": "Task Description",
                  "id": 1,
                  "title": "Task Title",
                },
              ]
            }
            renderListItem={[Function]}
          >
            <ListGroup
              bsClass="list-group"
              className="TasksList ShowTasks-TasksList "
            >
              <div
                className="TasksList ShowTasks-TasksList  list-group"
              >
                <TaskListItem
                  className=""
                  task={
                    Object {
                      "description": "Task Description",
                      "id": 1,
                      "title": "Task Title",
                    }
                  }
                >
                  <ListGroupItem
                    bsClass="list-group-item"
                    className="TaskListItem TaskListItem-1 "
                    listItem={false}
                  >
                    <span
                      className="TaskListItem TaskListItem-1  list-group-item"
                    >
                      <Row
                        bsClass="row"
                        componentClass="div"
                      >
                        <div
                          className="row"
                        >
                          <Col
                            bsClass="col"
                            className="TaskListItem-CheckBoxContainer"
                            componentClass="div"
                            md={2}
                            sm={2}
                          >
                            <div
                              className="TaskListItem-CheckBoxContainer col-md-2 col-sm-2"
                            >
                              <Checkbox
                                bsClass="checkbox"
                                disabled={false}
                                inline={false}
                                title=""
                              >
                                <div
                                  className="checkbox"
                                >
                                  <label
                                    title=""
                                  >
                                    <input
                                      disabled={false}
                                      type="checkbox"
                                    />
                                  </label>
                                </div>
                              </Checkbox>
                            </div>
                          </Col>
                          <Col
                            bsClass="col"
                            className="TaskListItem-TitleAndDescription"
                            componentClass="div"
                            md={8}
                            sm={8}
                          >
                            <div
                              className="TaskListItem-TitleAndDescription col-md-8 col-sm-8"
                            >
                              <Row
                                bsClass="row"
                                className="TaskListItem-TitleContainer"
                                componentClass="div"
                              >
                                <div
                                  className="TaskListItem-TitleContainer row"
                                >
                                  <Title
                                    className="TaskListItem-Title"
                                  >
                                    <h5
                                      className="Title TaskListItem-Title"
                                    >
                                      Task Title
                                    </h5>
                                  </Title>
                                </div>
                              </Row>
                              <Row
                                bsClass="row"
                                className="TaskListItem-DescriptionContainer"
                                componentClass="div"
                              >
                                <div
                                  className="TaskListItem-DescriptionContainer row"
                                >
                                  <Description
                                    className="TaskListItem-Description"
                                  >
                                    <p
                                      className="Description TaskListItem-Description"
                                    >
                                      Task Description
                                    </p>
                                  </Description>
                                </div>
                              </Row>
                            </div>
                          </Col>
                          <Col
                            bsClass="col"
                            className="TaskListItem-CheckBoxContainer"
                            componentClass="div"
                            md={2}
                            sm={2}
                          >
                            <div
                              className="TaskListItem-CheckBoxContainer col-md-2 col-sm-2"
                            >
                              <Button
                                active={false}
                                block={false}
                                bsClass="btn"
                                bsStyle="warning"
                                disabled={false}
                              >
                                <button
                                  className="btn btn-warning"
                                  disabled={false}
                                  type="button"
                                >
                                  Edit
                                </button>
                              </Button>
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </span>
                  </ListGroupItem>
                </TaskListItem>
              </div>
            </ListGroup>
          </List>
        </TasksList>
      </EmptyMessage>
    </ErrorMessage>
  </LoadingMessage>
</ListTasks>
`;
